app/cdap/components/CreateEditRecipeForm/index.tsx

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, { useEffect, useState } from 'react';
+import { FormControl } from '@material-ui/core';
+import T from 'i18n-react';
+import { ICommonRecipeFormProps, IRecipeData } from 'components/CreateEditRecipeForm/types';
+import {
+  FormFieldWrapper,
+  NormalLabel,
+  StyledForm,
+  StyledRecipeNameTextField,
+  StyledDescriptionTextArea,
+  StyledCancelButton,
+  StyledSaveButton,
+  StyledFormButtonWrapper,
+  getLabelStyle,
+} from 'components/CreateEditRecipeForm/styles';
+
+export default function({
+  recipeData,
+  onRecipeDataSave,
+  onCancel,
+  isNameError,
+  setIsNameError,
+}: ICommonRecipeFormProps) {
+  const StyledLabel = getLabelStyle(isNameError);
+  const [recipeFormData, setRecipeFormData] = useState<IRecipeData>({
+    recipeName: '',
+    description: '',
+    directives: [],
+  });
+
+  const [isSaveDisable, setIsSaveDisable] = useState<boolean>(true);
+
+  useEffect(() => {
+    if (isNameError) {
+      setIsSaveDisable(true);
+    }
+  }, [isNameError]);
+
+  useEffect(() => {
+    setRecipeFormData(recipeData);
+  }, [recipeData]);
+
+  useEffect(() => {
+    if (
+      recipeFormData.recipeName === '' ||
+      recipeFormData.description === '' ||
+      recipeFormData.recipeName?.trim().length === 0 ||
+      recipeFormData.description?.trim().length === 0
+    ) {
+      setIsSaveDisable(true);
+    } else {
+      setIsNameError(false);
+      setIsSaveDisable(false);
+    }
+  }, [recipeFormData]);
+
+  const onFormHandle = (event) => {
+    event.preventDefault();
+    onRecipeDataSave(recipeFormData);
+  };
+
+  return (
+    <>
```

Comment:
why we need extra fragment over styledform

Reply:
removed